smpclient/__init__.py

```diff
@@ -50,10 +56,18 @@ async def upload(
     ) -> AsyncIterator[int]:
         """Iteratively upload an `image` to `slot`, yielding the offset."""
 
+        max_packet_size = self._transport.max_unencoded_size
+        if isinstance(self._transport, SMPBLETransport):
+            mcumgr_parameters = await asyncio.wait_for(self.request(MCUMgrParametersRead()), timeout=1.0)
+            if success(mcumgr_parameters):
+                max_packet_size = mcumgr_parameters.buf_size
+
+        # why 23? should be max_packet_size
         if self._transport.max_unencoded_size < 23:
             raise Exception("Upload requires an MTU >=23.")
```

Comment:
Having a heck of a time with the UART transport's encoding requirements.  The unit tests may make it more clear what going on.

Reply:
For ble 23 is supported if reassemble is enabled.

Reply:
OK, I think it is probably safe to remove this exception.